### ISSUE TITLE
avoid_annotating_with_dynamic's description only mentions return type.

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#3941]
**Describe the issue**

The avoid_annotating_with_dynamic description says (only):

> **AVOID** annotating with dynamic when not required.
>
> As `dynamic` is the assumed return value of a function or method, it is usually
not necessary to annotate it.

Then these are the nodes it registers:

```dart
    registry.addFieldFormalParameter(this, visitor);
    registry.addSimpleFormalParameter(this, visitor);
    registry.addSuperFormalParameter(this, visitor);
```

Uh... wait what? The description makes it sound (perhaps) like return types are the concern here. Is this rule for like... variable declarations? Only parameters? Only return types? Anywhere a type annotation can be found?

(While we're at it, the sentence about return types is super false, and should be rewritten, if the rule should continue to say something about return types.)

**Comment by bwilkerson** [comment]
> Anywhere a type annotation can be found?

I believe that that was the original intent of the lint, though I might be misremembering on this one.

But I have to wonder whether that's a style we actually want to support in the null-safe world. We might want to consider deprecating this one.

**Comment by srawlins** [comment]
I would be very on board with deprecating this one. :) It is _only_ flagging parameters, and I don't even think it catches parameters with default values. It seems pretty arbitrary to me, and not motivated by any style guide.
